apps/client/src/components/HtmlViewer/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect } from 'react'
+
+const HTMLViewer = ({ html }: { html: string }) => {
+  useEffect(() => {
+    const tempDiv = document.createElement('div')
+    tempDiv.innerHTML = html
+
+    let currentNonce = ''
+    const injectedIdentifier: string[] = []
+    const scripts = tempDiv.querySelectorAll('script')
+
+    scripts.forEach((script, index) => {
+      const newScript = document.createElement('script')
+      const scriptNonce = script.getAttribute('nonce')
+      if (scriptNonce && !currentNonce) {
+        currentNonce = scriptNonce
+      }
+
+      if (script.src) {
+        newScript.src = script.src
+      } else {
+        newScript.textContent = script.textContent
+      }
+
+      const identifier = `scraper-${index}`
+      newScript.setAttribute('temp-script', identifier)
+
+      document.head.appendChild(newScript)
+
+      injectedIdentifier.push(identifier)
+    })
+
+    return () => {
+      injectedIdentifier.forEach((identifier) => {
+        const scriptToRemove = document.querySelector(`[temp-script="${identifier}"]`)
+        if (scriptToRemove?.parentNode) {
+          scriptToRemove.parentNode.removeChild(scriptToRemove)
+        }
+      })
+
+      const scriptElements = document.querySelectorAll('script')
+      scriptElements.forEach((script) => {
+        const nonceValue = script.getAttribute('nonce')
+        if (nonceValue === currentNonce && script?.parentNode) {
+          script.parentNode.removeChild(script)
+        }
+      })
+
+      tempDiv.remove()
+    }
+  }, [html])
+
+  return <div dangerouslySetInnerHTML={{ __html: html }} />
+}
+
+export default HTMLViewer
```

Comment:
Somehow, the `dangerouslySetInnerHTML` approach is not working well with inline scripts, and this issue is affecting the rendering of images within the DOM

To address this issue, I came up with injecting scripts, which seems to have resolved the issue.

I also attempted to use a different library that sanitizes HTML content, but the output appears to be different. It seems that the sanitization process removed a significant portion of the scripts, resulting in unexpected output